packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -118,6 +118,29 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         );
         expect(getColumnValues()).to.deep.equal(['Jon Snow', 'Cersei Lannister']);
       });
+
+      it('should support columns.valueGetter even if field param supplied is incorrect resulting in no data to be rendered', () => {
+        const columns = [
+          { field: 'id', hide: true },
+          { field: 'firstName', hide: true },
+          { field: 'lastName', hide: true },
+          {
+            field: 'fullName',
+            valueGetter: (params) => params.getValue('age'), // incorrect field parameter supplied to 'getValue'
```

Comment:
I think we should add a warning or throw an error as it is failing silently atm, which is not ideal

Reply:
@dtassone We need your guidance on this one. The comment is not synched with the implementation. We went in the direction of making the above fully supported by the grid. Is this behavior in the test something we want to support or would you rather not support it? If we don't want to support it, I would say adding a warning with instruction to resolve it would be great.

Reply:
I will udpate the with the direction proposed by Damien. Basically, we don't support this use case, we want to warn.